### PR TITLE
Docs: Add Schema maturity docs

### PR DESCRIPTION
**Pull request by rhorvath**
**What is this feature?**

It adds 'Grafana Kinds - From Zero to Maturity' into the docs. 
This doc is then referenced from auto-generated Kind docs for Schema Maturity definition lookup.

**Why do we need this feature?**

It improves understanding of generated Kinds and their maturity levels.  

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/schematization-and-as-code-project/issues/49

**Special notes for your reviewer**:
It builds on top of https://github.com/grafana/grafana/pull/60416

### CODE REVIEW

File: docs/sources/developers/kinds/maturity.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+[*never* produce stubs](https://github.com/grafana/grafana/pull/54816#issuecomment-1256341790). Consequently, it is a 
+system you can learn a lot about through trial and error: try something, run the generator, then see the effects by 
+running `git diff`.
+
+When writing Thema schemas within Grafana, there are a few key differences vs. the approach in the Thema tutorial:
+- All schema-based codegen in grafana/grafana is performed by make `gen-cue`, rather than by invoking the thema CLI.
+- Grafana code generators support certain attributes the Thema CLI is not natively aware of.
+- Thema lineages are expected to be declared at specific subpaths within .cue files, varying by category.
+
+## Schema-writing guidelines
```
Comment (sdboyer) [inline docs/sources/developers/kinds/maturity.md:109]: Let's separate these out into their own doc in a follow-up PR

File: docs/sources/developers/kinds/maturity.md [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+But, that documentation can only be as good as the source from which it’s generated. This section offers guidance 
+on how to write effective documentation for kinds in the kind declarations themselves.
+
+### Kind-level documentation
+TODO providing a high-level summary of the type
+TODO link to examples of where it shows up
+
+### Field-level documentation
+TODO criteria for writing good field-level docs
+TODO link to examples of where it shows up**
```
Comment (sdboyer) [inline docs/sources/developers/kinds/maturity.md:240]:
```suggestion
```

File: docs/sources/developers/kinds/maturity.md [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+| **Common hurdles**           | (We don’t know yet!)                                                                                                                                                                                                                                                                                                                      |
+| **Public-facing guarantees** | Backward compatibility for newer versions with previous schema versions (starting from open beta, 1.0 version). General support agreement for our GA features                                                                                                                                                                             |
+| **User-facing stage**        | Beta - open: anyone can use schemas, the functionality is marked asd beta.<br/> - Define metrics for Beta to GA transition (adoption? Number of change requests and bug opened trending down? ) Or should we skip open beta? All suggestions welcomeGA: the schema is launched and can be used as part of the Grafana toolkit for as-code |
+
+
```
Comment (sdboyer) [inline docs/sources/developers/kinds/maturity.md:307]:
```suggestion
```

i think we should omit these for now - the definition of a maturity stage is an important thing to have formal agreement on by our arch board

File: docs/sources/developers/kinds/maturity.md [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+# Schema Attributes
```
Comment (sdboyer) [inline docs/sources/developers/kinds/maturity.md:308]: Fine to be here for now, but these also deserve their own doc

File: docs/sources/developers/kinds/maturity.md [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+
+Primarily useful on very large schemas, like the dashboard schema, for getting *something* written down for a given 
+field that at least makes validation tests pass, but making clear that the field isn’t necessarily properly correct.
+
+No argval is accepted. (Use a `//` comment to say more about the attention that’s needed.)
+
+### MaybeRemove
+
+Field was added as part of applying the maximalist rule, and should be considered for removal in a subsequent review of 
+the schema.
```
Comment (sdboyer) [inline docs/sources/developers/kinds/maturity.md:403]:
```suggestion
```
not clear we actually want this in a k8s object-centric world

**Review by sdboyer — DISMISSED** [review]

File: docs/sources/developers/kinds/maturity.md (hunk 2 above)
Comment (jdbaldry) [reply docs/sources/developers/kinds/maturity.md:240]: > **TODO oof is this a complicated topic**
:joy:

Comment (sdboyer) [reply docs/sources/developers/kinds/maturity.md:240]: yeah, and i have to write a quick treatment of it today, in another doc :) (one we'll eventually move out to be public-facing, like these)

File: docs/sources/developers/kinds/maturity.md [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+for every large-scale refactoring.
+
+**This document is the canonical reference for *how* to write Grafana schemas: start from nothing, and confidently 
+iterate to a mature initial version.** (Maintaining and evolving schemas *after* maturity is reached is a topic for 
+another doc.)
+
+There are several planned categories of Kinds for Grafana. The schema authoring workflow for each varies, as does the 
+path to maturity. This doc contains general reference material applicable to all Kind-writing, and links to the guides 
+for each category of Kind. Shortly, we’ll give you the necessary information to figure out which guide to follow. Before
+that, though, some context on the overarching goal of “schema maturity” is helpful.
```
[the remark this answers is not in the document]
Comment (rhorvath) [reply docs/sources/developers/kinds/maturity.md]: Good point, I took it from the maturity docs and moved it into the parent (slightly simplified).

File: docs/sources/developers/kinds/maturity.md (hunk 1 above)
Comment (rhorvath) [reply docs/sources/developers/kinds/maturity.md]: Agreed, in the follow-up PRs. For the moment I pushed it below the Milestones section. Hope it's fine

**Comment by sdboyer** [comment]
can we...still do this, while schema day is running? 😬

**Review by undef1nd — APPROVED** [review]

**Comment by sdboyer** [comment]
@rhorvath we should be able to wrap this one up now, yeah?

**Review by joanlopez — APPROVED** [review]
LGTM!

**Review by malcolmholmes — APPROVED** [review]

**Pull request merged**